packages/xstate-react/README.md

```diff
@@ -252,6 +267,22 @@ const App = ({ service }) => {
 };
 ```
 
+With Typescript, you might want to define the selector and comparator inline for better type inference. In that case, you still need to use `React.useCallback()` for both. To get proper types you need to provide the 2 generic arguments to `useSelector()` - the type of the service we're selecting from, and the output type of our selector.
```

Comment:
when using inline functions, like here - shouldn't types be just inferred properly by TS?